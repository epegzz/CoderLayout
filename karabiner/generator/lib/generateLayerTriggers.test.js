const generateLayerTriggers = require('./generateLayerTriggers')

describe('generateLayerTriggers', () => {
  it('works', () => {
    const triggerKeys = ['a', 'b']
    expect(generateLayerTriggers(triggerKeys)).toMatchSnapshot()
  })
})
