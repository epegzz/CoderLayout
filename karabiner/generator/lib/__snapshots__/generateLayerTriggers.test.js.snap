// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLayerTriggers works 1`] = `
Object {
  "description": "Layer triggers",
  "manipulators": Array [
    Object {
      "from": Object {
        "key_code": "a",
        "modifiers": Object {
          "optional": Array [
            "any",
          ],
        },
      },
      "to": Array [
        Object {
          "set_variable": Object {
            "name": "trigger_a",
            "value": 1,
          },
        },
      ],
      "to_after_key_up": Array [
        Object {
          "set_variable": Object {
            "name": "trigger_a",
            "value": 0,
          },
        },
      ],
      "type": "basic",
    },
    Object {
      "from": Object {
        "key_code": "b",
        "modifiers": Object {
          "optional": Array [
            "any",
          ],
        },
      },
      "to": Array [
        Object {
          "set_variable": Object {
            "name": "trigger_b",
            "value": 1,
          },
        },
      ],
      "to_after_key_up": Array [
        Object {
          "set_variable": Object {
            "name": "trigger_b",
            "value": 0,
          },
        },
      ],
      "type": "basic",
    },
  ],
}
`;
